### PR TITLE
PostgreSQL connector support, based on new CloudFoundry service bindings

### LINKED ISSUE #199: Add support for Postgres supplied via GCP service broker
Postgres instances created by the GCP service broker require client certificate configuration and support for ssl-enabled connections. Steeltoe already provides support for passing in an action that can be used (in a very clunky way) to provide the certificate, but also needs to support setting `SslMode.Require` based on the service binding

### PR DESCRIPTION
**Pull request by bart-vmware**
## Description
Builds on top of the CloudFoundry service bindings, providing IOptions support and health checks.

Fixes #199.

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [x] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: src/Connectors/src/Connector/PostgreSql/PostgreSqlWebApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+using Steeltoe.Common;
+using Steeltoe.Common.HealthChecks;
+
+namespace Steeltoe.Connector.PostgreSql;
+
+public static class PostgreSqlWebApplicationBuilderExtensions
+{
+    private const string BindingType = "postgresql";
+
+    public static WebApplicationBuilder AddPostgreSql(this WebApplicationBuilder builder)
```
Comment (hananiel) [inline src/Connectors/src/Connector/PostgreSql/PostgreSqlWebApplicationBuilderExtensions.cs:21]: Why is this the only connector with a webApplicationBuilder extension? Do we in addition need a Iservicecollection.AddPostgreSql?

Comment (bart-vmware) [reply src/Connectors/src/Connector/PostgreSql/PostgreSqlWebApplicationBuilderExtensions.cs:21]: Yes, definitely. Thanks for pointing this out. It's on my list of enhancements to address later. This PR is the bare minimum to make our PostgreSQL sample work, without changing the existing connectors' logic. That's also why I had to add a few ***2 types, to be renamed later.

Once I've added support for the other connectors, I'm going to refactor, clean up, document, etc.

**Comment by bart-vmware** [comment]
The packaging task currently fails due to a new bug in the .NET SDK. It should be gone once the cloud-hosted Azure DevOps servers have been updated to v7.0.201. See https://github.com/dotnet/sdk/issues/30625#issuecomment-1445175355.
```
C:\Program Files\dotnet\sdk\7.0.200\Current\SolutionFile\ImportAfter\Microsoft.NET.Sdk.Solution.targets(36,5): error NETSDK1194: The "--output" option isn't supported when building a solution. [D:\a\1\s\src\Steeltoe.All.sln]
                            ^^^^^^^
```

**Comment by TimHess** [comment]
Will this resolve #199?

**Comment by bart-vmware** [comment]
> Will this resolve #199?

Yes, I've added it to the PR description.

**Pull request merged**